File: common/src/main/java/com/harera/hayat/common/repository/GlobalMessageRepository.java
package com.harera.hayat.common.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.harera.hayat.common.model.exception.GlobalMessage;

@Repository
public interface GlobalMessageRepository extends JpaRepository<GlobalMessage, Long> {
    Optional<GlobalMessage> findByLanguageAndCode(String language, String code);
}

File: src/main/java/com/harera/dwaaserver/repository/GlobalMessageRepository.kt
package com.harera.dwaaserver.repository;

import com.harera.dwaaserver.model.GlobalMessage
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface GlobalMessageRepository : JpaRepository<GlobalMessage, Long> {

    fun findByLanguageAndCode(language: String?, code: String?): Optional<GlobalMessage>
}
